### PR TITLE
bsc#988980: Add ssl option to cinder.conf

### PR DESCRIPTION
**Pull request by nicolasbock**

### CODE REVIEW

**Comment by nicolasbock** [comment]
Triggered gating rebuild.

**Comment by rsalevsky** [comment]
Can you trigger a CI run with both PR's to be sure it really works?

**Comment by nicolasbock** [comment]
Sure, but how do I combine two PRs in ci?

File: chef/cookbooks/cinder/templates/default/cinder.conf.erb [hunk 1]
```diff
@@ -580,6 +580,11 @@ osapi_volume_listen = <%= @bind_host %>
 #osapi_volume_listen_port = 8776
 osapi_volume_listen_port = <%= @bind_port %>
 
+# Wraps the socket in a SSL context if True is set. A certificate file and key
+# file must be specified.
+#osapi_volume_use_ssl = False
```
Comment (aplanas) [inline chef/cookbooks/cinder/templates/default/cinder.conf.erb]: Is this option missing in the generated config file? If so, we need to add a note and maybe fix it upstream (or point to the review that is now on going)

Comment (nicolasbock) [inline chef/cookbooks/cinder/templates/default/cinder.conf.erb]: Let me check that.

**Comment by rsalevsky** [comment]
In the automation PR there is gating with both PRs.

**Comment by rsalevsky** [comment]
Triggert a rebuild. The magnum error is fixed.

File: chef/cookbooks/cinder/templates/default/cinder.conf.erb (hunk 1 above)
Comment (nicolasbock) [inline chef/cookbooks/cinder/templates/default/cinder.conf.erb]: @aplanas We were missing this option in our conf, upstream has it. I generated a new one from https://github.com/openstack/cinder/tree/stable/mitaka and compared it to ours. I have added another commit to this PR that simply updates the configuration file, but doesn't add the ssl option to it. I left that for the second commit.

**Comment by nicolasbock** [comment]
> In the automation PR there is gating with both PRs.

@rsalevsky Sorry, but I don't see it. Looking at the gating of https://github.com/SUSE-Cloud/automation/pull/1296, I find `github_pr` set to one value. And looking at https://github.com/SUSE-Cloud/automation/blob/master/scripts/jenkins/ci.suse.de/openstack-mkcloud.xml#L413 it looks like that variable is expected to contain only one PR. Could you explain in more detail what you meant?

**Comment by nicolasbock** [comment]
https://github.com/SUSE-Cloud/automation/issues/1296 was merged. Triggered rebuild with `ssl`.

**Comment by nicolasbock** [comment]
Build is failing in glance_v{1,2} which should be fixed by https://github.com/crowbar/crowbar-openstack/pull/547. Additional failures in manila, tracked here:

- https://bugzilla.suse.com/show_bug.cgi?id=1003923
- https://bugzilla.suse.com/show_bug.cgi?id=1003924

**Comment by nicolasbock** [comment]
To me the failures here look unrelated to the PR. But of course it would be nice if the test actually passed :(

**Comment by nicolasbock** [comment]
Rebased.

**Comment by aplanas** [comment]
Gating fails in manila SSL (tempest):

* `manila_tempest_tests.tests.api.admin.test_share_networks.ShareNetworkAdminTest`
  `SSLHandshakeError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed`

* `manila_tempest_tests.tests.api.test_share_networks.ShareNetworksTest`
   `SSLHandshakeError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed`

@toabctl any clue?

**Comment by vuntz** [comment]
@nicolasbock I'm nearly sure your rebased template is from newton; a good way to see that in the mention of "castellan", which is not in mitaka.

**Comment by vuntz** [comment]
FWIW, I'd go with only the first commit; the manila issues are unrelated imho, they would just be the next thing to fix.

**Comment by aplanas** [comment]
better now!

**Comment by nicolasbock** [comment]
Only manila failing.

**Comment by nicolasbock** [comment]
@rsalevsky what do you think?

**Pull request merged**